python/cudf/cudf/dataframe/column.py

```diff
@@ -435,12 +434,14 @@ def element_indexing(self, index):
         ------
         ``IndexError`` if out-of-bound
         """
-        index = int(index)
+        index = np.int32(index)
         if index < 0:
             index = len(self) + index
         if index > len(self) - 1:
             raise IndexError
         val = self.data[index]  # this can raise IndexError
+        if isinstance(val, nvstrings.nvstrings):
+            val = val.to_host()[0]
```

Comment:
given that `to_host()` copies potentially an entire list of strings to the CPU, if we expect to at some point support arrays of strings as a column datatype, does it make sense to break this out into two cases, one for `['some_string']` and another for `['some', 'potential', 'list', 'of', 'strings']`?

Reply:
Great point - although I'm not sure what I can change here at the moment. There's definitely other parts of the code that assume a column of single values and specializing here won't be enough.

FWIW I think indexing into an element always triggers a copy to host (not just for strings)...